runtime/connectors/s3/s3.go

```diff
@@ -76,3 +87,66 @@ type connector struct{}
 func (c connector) Spec() connectors.Spec {
 	return spec
 }
+
+// Consume will eventually be added to the interface
+func Consume(ctx context.Context, source *connectors.Source) (string, error) {
+	conf, _ := ParseConfig(source.Properties)
+
+	// The session the S3 Downloader will use
+	sess := session.Must(session.NewSession(&aws.Config{
+		Region:      &conf.AWSRegion,
+		Credentials: getAwsCredentials(conf),
+	}))
+
+	// Create a downloader with the session and default options
+	downloader := s3manager.NewDownloader(sess)
+
+	bucket, key, filename, err := getAwsUrlParts(conf.Path)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse path %s, %v", conf.Path, err)
+	}
```

Comment:
Like in GCS, what if bucket is set in the `conf`? Should we disable that option?

Reply:
Similarly the bucket is part of the url from `Path`.